src/readline.ts

```diff
@@ -315,6 +315,9 @@ export class Readline implements ITerminalAddon {
       case InputType.CtrlU:
         this.state.update("");
         break;
+      case InputType.CtrlK:
+        this.state.editDelete(this.state.buffer().length);
```

Comment:
Using `this.state.buffer().length` will often be a larger value than necessary, but I did not see a simple way to compute the exact number of characters to delete to the end of the line, and this works correctly in my testing.

Another possibility is to modify the `State` class to do a more precise operation.

Reply:
Yeah, this would end up deleting multiple lines in multi line code blocks. I think that adding this to line.ts:

```js
  public deleteEndOfLine(): boolean {
    if (this.buf.length == 0 || this.pos == this.buf.length) {
      return false;
    }

    const start = this.pos;
    const end = this.endOfLine();
    if (start == end) {
      this.delete(1);
    } else {
      this.buf = this.buf.slice(0, start) + this.buf.slice(end);
    }

    return true;
  }```
  
  And then adding a supporting function in state.ts called by the ctrl-k code would do it:
  
  ```js
    public editDeleteEndOfLine() {
    if (this.line.deleteEndOfLine()) {
      this.refresh();
    }
  }```